isaac2/Assets/Scripts/EnemyPursue.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyPursue : MonoBehaviour
+{
+    /// <summary>
+    /// Enemies pursue Player when alert.
+    /// Enemies may keep a set following distance.
+    /// 
+    /// NOTES:
+    /// Pursue Player
+    ///     - Stay in place, never pursue
+    ///     - Pursue when alert
+    ///     - Following distance
+    ///         - Always try to touch Player (melee & explosive attacks, etc.)
+    ///         - Keep a set distance away (ranged attacks, etc.)
+    ///             - Flee when Player gets closer than following distance
+    ///             - Allow Player to get close
+    ///     - Pathing
+    ///         - Elements of randomness
+    ///         - Avoid obstacles & dangers
+    ///         - Avoid walls & navigate corners
+    /// Flee Player
+    ///     - Enemies could flee under certain conditions (low health, ranged attacks)
+    /// Speed
+    ///     - Not the current velocity, but how "fast" an enemy is?
+    ///     - Base speed
+    ///         - Set on spawn
+    ///         - Different types of enemy, different speeds
+    ///         - Randomly selected from range
+    ///     - Speed
+    ///         - Can be changed by powerups/effects
+    ///         
+    /// </summary>
+
+
+    private GameObject player;
+    [SerializeField] private float baseSpeed;
+    [SerializeField] private float speed;
+    private bool isAlert;
+    [SerializeField] private float followingDistance;  // amount of distance from player an enemy would like to be
+    private float distance;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = GameObject.FindGameObjectWithTag("Player");
+        isAlert = this.GetComponent<EnemyDetectPlayer>().getIsAlert(); 
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        distance = Vector2.Distance(transform.position, player.transform.position);
+
+        isAlert = this.GetComponent<EnemyDetectPlayer>().getIsAlert();
+        if (isAlert && distance > followingDistance)
+        {
+            transform.position = Vector2.MoveTowards(this.transform.position, player.transform.position, speed * Time.deltaTime);
```

Comment:
please make it so enemies stop moving when they are dead.

Reply:
Done. Wouldn't it be best to destroy gameobject on death and replace with a no-collision no-AI "corpse" object?